### DISCUSSION TITLE
Can't Connect to GDS 2.0

### DISCUSSION THREAD

**Discussion by legounicycler** [nasa/fprime#1280] [Q&A] [answered]
Hello all,

My goal is to use the new FPrime GDS with the old GroundInterface component to relay commands, events, and telemetry. I'm having trouble connecting to the GDS with my deployment, and I'm not quite sure how to proceed from where I am. Let me break down the setup and what I have done so far.

I am working with fprime v2.0.1. I installed the GDS by cloning the fprime-gds repository [here](https://github.com/fprime-community/fprime-gds), checking out the 2.0 tag, and running `pip install -e .` The deployment I have built is running on an ATMega128, and it uses the GroundInterface component to frame and deframe the packets sent and received. I know that this component has been replaced by the Framer/Deframer components, but I am using it because I am attempting to upgrade an old deployment (which was built for Fprime 1.3) to be compatible with Fprime 2.0.1. The ground interface component is connected to an ATmega UART driver running at 115200 baud that sends data over a USB cable to my computer.

On the GDS end of things, I started the GDS with the following command:
`fprime-gds --comm-adapter uart --dictionary ./Top/LEDBlinkDeploymentTopologyAppDictionary.xml --uart-device /dev/ttyUSB0 --uart-baud 115200 --comm-checksum-type fixed --no-app`

When sprinkling print statements throughout the GDS source code, I've verified that the data is being received through the UART adapter, successfully getting deframed, passing the checksum, and being sent to the TCP server. However, I've determined by adding print statements that the packets are never reaching the distributor. This is where my knowledge gets shaky, because I know very little about networking. If my understanding is correct, the data is either not making it from the TCP server to the TCP client, or it is not making it from the TCP client to the distributor. I've traced my way all the way through the code until reaching the `write_impl()` function in the `ip.py` file, where I verified that `IpHandler.CONNECTED` is connected. After this, I'm not sure where the data is going, and all I can figure out is that it's not being received by the distributor. I printed out the value of `self.client` which shows up as the following:
```
<socket.socket fd=3, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('127.0.0.1', 53628), raddr=('127.0.0.1', 50050)>
```
I used wireshark to determine that there are indeed packets flowing between 127.0.0.1 port 50050 and 127.0.0.1 port 53630.

At this point, I've done a lot of work to diagnose the problem, but I feel like I'm missing something simple. Is there some documentation I can be looking at?

Thanks for the help.

**Comment by Joshua-Anderson** [comment]
@legounicycler Try setting `--comm-checksum-type=fixed`. The old groundinterferance component wasn't checksumming F' packets. The newer version of the GDS assumes packets are CRC32 checksummed and drops packets with invalid checksums.

**Reply by legounicycler** [reply]
Unless there is somewhere else to specify things, as you can see above, I did include this flag in my fprime-gds command, and I verified that the checksum is valid in the deframing portion of the GDS. Unless I am misunderstanding the checksum somehow, this should not be the issue.

**Comment by LeStarch** [comment]
If the call is getting to the TCP server, then it is past the stage where check summing is done.  I've never seen an issue with the `ThreadedTcpServer` so I would be surprised if it was not passing data through.

Did you see any issues in the log files for the GDS?  There might be an error in there somewhere. I am guessing there was trouble allocating port 50050 and the error is in the log files somewhere (e.g. REf/log/<timestamp>/)

In addition, you can use the latest GDS with older versions of F´.  Have you tried `pip install -U fprime-gds` and running the above command?

The next place to place a print statement if those don't help is:

https://github.com/fprime-community/fprime-gds/blob/5170499320e64b51a037c75fb4a0a02e7cc89aa2/src/fprime_gds/common/distributor/distributor.py#L180

To show the on_recv method is called within the GDS.

**Comment by legounicycler** [comment, marked as answer]
It's been a while since I came back to this post, but just wanted to update that I indeed did get this issue resolved. It turns out that the packets WERE traveling through the TCP connection as expected, and I was just missing the debug print statement I had used to verify this as it got burried under lots of other debug print statements. The ultimate error I had made was that I changed the packet header types from F32 to F16 in my FPrime deployment, but hadn't changed the configuration of the GDS to accept these smaller packet headers.
